Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
-#include "../include/Volunteer.h";
+
+#include "../include/Volunteer.h"
+#include "../include/Order.h"
+
+
+/////////////////////////////////////////////////////////
+//////////////////////// Volunteer //////////////////////
+/////////////////////////////////////////////////////////
+
+Volunteer::Volunteer(int id, const string &name) : id(id), name(name) 
+{
+    completedOrderId = -1;
+    activeOrderId = -1;
+}
+
+Volunteer::Volunteer(int id, const string &name, int completedOrderId, int activeOrderId) : id(id), name(name), completedOrderId(completedOrderId), activeOrderId(activeOrderId) {}
+
+int Volunteer::getId() const
+{
+    return id;
+}
+
+const string& Volunteer::getName() const 
+{
+    return name;
+}
+
+int Volunteer::getActiveOrderId() const 
+{
+    return activeOrderId;
+}
+
+bool Volunteer::isBusy() const
+{
+    return (activeOrderId != NO_ORDER);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// Collector Volunteer ////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+CollectorVolunteer::CollectorVolunteer(int id, const string &name, const int coolDown) : Volunteer(id, name) ,coolDown(coolDown), timeLeft(NO_ORDER){}
```

Comment:
time left maybe should be 0 and no NO_ORDER

Reply:
i think it have to be according to the decreaseCooldown

Reply:
if so, decreaseCooldown() will always return 'true'

Reply:
Correct me if i'm wrong

Reply:
you decrease only after accept order, so it will be updated.
notice that it will be 0 after the first order finished

Reply:
Right <3